File: 1319-unique-number-of-occurrences/unique-number-of-occurrences.cpp
class Solution {
public:
    bool uniqueOccurrences(vector<int>& arr) {
        vector<int> poshash(2002,0);
        vector<int> hash(1001,0);
        for (int i = 0;i < arr.size();i++)
        {
            if (arr[i] < 0)
                poshash[(arr[i] * -1) + 1001]++;
            else 
                poshash[arr[i]]++;
        }
        for (int i = 0;i < poshash.size();i++)
            if (poshash[i] > 0)
                hash[poshash[i]]++;
        for (int i = 0; i < hash.size();i++)
            if (hash[i] >= 2)
                return false;
        return true;
    }
};

File: 1319-unique-number-of-occurrences/unique-number-of-occurrences.py
class Solution(object):
    def uniqueOccurrences(self, arr):
        """
        :type arr: List[int]
        :rtype: bool
        """
        dics = {}
        for n in arr:
            if n in dics:
                dics[n] += 1
            else:
                dics[n] = 1
        return len(dics) - len(set(dics.values())) == 0

        
            
        